pythonPackages.allure-pytest: init at 2.9.45

evanjs opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
## Add `allure-pytest` to nixpkgs
## Add `allure-python-commons` to nixpkgs 
## Add `allure-python-commons-test` to nixpkgs 

###### Notes
~Tests are currently disabled.~
A few integration tests are being skipped.

I am not familiar with how `tox` tests are usually handled, but some lazy grepping shows some packages removing `tox.ini` completely, etc.

See [this issue](https://github.com/allure-framework/allure-python/issues/438) for more information

___
## Allure Python packages
I only added `allure-pytest` and `allure-python-commons`
There is also e.g. `allure-python-commons-test`, `allure-pytest-bdd`, etc.


I have added the packages required to get Allure reporting working for Pytest, and it seems to be collecting valid results, which are rendered nicely on Jenkins using the associated Allure plugin.

![image](https://user-images.githubusercontent.com/1847524/138602419-2d1f101b-51da-4c32-8538-e8c63800b6aa.jpeg)

___

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

cc @addict3d

SuperSandro2000 commented on pkgs/development/python-modules/allure-python-commons/default.nix:
```diff
@@ -0,0 +1,27 @@
+      --replace "setup_requires=['setuptools_scm']," "setup_requires=[],"
+  '';
+
+  buildInputs = [ attrs pluggy six ];
```
This needs to be propogatedBuildInputs.

SuperSandro2000 commented on pkgs/development/python-modules/allure-python-commons/default.nix:
```diff
@@ -0,0 +1,27 @@
+  pname = "allure-python-commons";
+  version = "2.9.45";
+
+  disabled = isPy27;
```
Please use pythonOlder.

SuperSandro2000 commented on pkgs/development/python-modules/allure-pytest/default.nix:
```diff
@@ -0,0 +1,36 @@
+    sed 's;"setuptools_scm";;' -i setup.py
+  '';
+
+  buildInputs = [ six pytest ];
```
Why is six not propogated?

SuperSandro2000 commented on pkgs/development/python-modules/allure-pytest/default.nix:
```diff
@@ -0,0 +1,36 @@
+  };
+
+  postPatch = ''
+    sed 's;"setuptools_scm";;' -i setup.py
```
Please use substituteInPlace.

SuperSandro2000 commented on pkgs/development/python-modules/allure-pytest/default.nix:
```diff
@@ -0,0 +1,36 @@
+  pname = "allure-pytest";
+  version = "2.9.45";
+
+  disabled = isPy27;
```
Please use pythonOlder.

SuperSandro2000 commented on pkgs/development/python-modules/allure-python-commons/default.nix:
```diff
@@ -0,0 +1,27 @@
+  '';
+
+  propagatedBuildInputs = [ attrs pluggy six ];
+
```
Are there any tests we can run? Please add a pythonImportsCheck.

SuperSandro2000 commented on pkgs/development/python-modules/allure-pytest/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  # Tests require `-p pytester {posargs} for use with tox
+  doCheck = false;
+
```
Please add a pythonImportsCheck.

evanjs committed:
pythonPackages.allure-python-commons-test: init at 2.9.45

evanjs committed:
pythonPackages.allure-python-commons: init at 2.9.45

evanjs committed:
pythonPackages.allure-pytest: init at 2.9.45

evanjs commented:
I added `allure-python-commons-test` and enabled the acceptance tests for `allure-pytest`.

Integration tests are being skipped. 
By skipping this test suite, we are excluding 14 passed and 3 failed tests, as far as I can tell.

This brings us to 188 passed, 6 skipped, and 6 xfailed tests for `allure-pytest` by only running the acceptance test suite.

`allure-python-commons` and `allure-python-commons-test` are now both running doctests, but will not print results unless tests fail.
I am not sure if this is ideal or if we always want to print test results, etc.

I have also added `pythonImportsCheck` to each package.

SuperSandro2000 commented on pkgs/development/python-modules/allure-python-commons/default.nix:
```diff
@@ -0,0 +1,49 @@
+
+  propagatedBuildInputs = [ attrs pluggy six allure-python-commons-test ];
+
+  checkInputs = [
+    flake8
+  ];
+
+  checkPhase = ''
+    python -m doctest ./src/utils.py
+    python -m doctest ./src/mapping.py
+
+    flake8 src/
+  '';
```
```suggestion
  checkPhase = ''
    ${python.interpreter} -m doctest ./src/utils.py
    ${python.interpreter} -m doctest ./src/mapping.py
  '';
```
Running lint tests is only prone to new errors when we update flake8 but does not ensure the code is functioning correct.

SuperSandro2000 commented on pkgs/development/python-modules/allure-python-commons/default.nix:
```diff
@@ -0,0 +1,49 @@
+    sha256 = "17alymsivw8fs89j6phbqgrbprasw8kj72kxa5y8qpn3xa5d4f62";
+  };
+
+  buildInputs = [
```
```suggestion
  nativeBuildInputs = [
```
setuptools-scm is run on the host.

SuperSandro2000 commented on pkgs/development/python-modules/allure-python-commons-test/default.nix:
```diff
@@ -0,0 +1,44 @@
+  propagatedBuildInputs = [ attrs pluggy six pyhamcrest ];
+
+  checkPhase = ''
+    python -m doctest ./src/container.py
+    python -m doctest ./src/report.py
+    python -m doctest ./src/label.py
+    python -m doctest ./src/result.py
```
```suggestion
    ${python.interpreter} -m doctest ./src/container.py
    ${python.interpreter} -m doctest ./src/report.py
    ${python.interpreter} -m doctest ./src/label.py
    ${python.interpreter} -m doctest ./src/result.py
```

SuperSandro2000 commented on pkgs/development/python-modules/allure-python-commons-test/default.nix:
```diff
@@ -0,0 +1,44 @@
+    sha256 = "0rn8ccxxrm27skv3avdiw56zc4fk2h7nrk3jamqmx6fnvmshiz5f";
+  };
+
+  buildInputs = [
```
```suggestion
  nativeBuildInputs = [
```

SuperSandro2000 commented on pkgs/development/python-modules/allure-pytest/default.nix:
```diff
@@ -0,0 +1,76 @@
+    --alluredir=$allureDir \
+    -W ignore::pytest.PytestExperimentalApiWarning \
+    -p pytester ./test/acceptance
+
+    flake8 src/ test/
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/allure-pytest/default.nix:
```diff
@@ -0,0 +1,76 @@
+  checkPhase = ''
+    baseTemp=$(mktemp -d)
+    allureDir=$(mktemp -d allure-results.XXXXXXX)
+    echo "TMP directory: $baseTemp"
+    echo "Allure directory: $allureDir"
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/allure-pytest/default.nix:
```diff
@@ -0,0 +1,76 @@
+    echo "TMP directory: $baseTemp"
+    echo "Allure directory: $allureDir"
+
+    # we are skipping integration tests for now
+    # the only suite remaining is acceptance tests
+    py.test --basetemp=$baseTemp \
+    --alluredir=$allureDir \
+    -W ignore::pytest.PytestExperimentalApiWarning \
+    -p pytester ./test/acceptance
```
Please use pytestCheckHook with pytestFlagsArray and pytestDisaledTests.

SuperSandro2000 commented on pkgs/development/python-modules/allure-pytest/default.nix:
```diff
@@ -0,0 +1,76 @@
+    sha256 = "18ys5gi64jlfya6a7shj5lqhwc6cplwgyq3s2n5mg5x513g0yqi0";
+  };
+
+  buildInputs = [
+    pytest
+    setuptools-scm
+  ];
```
```suggestion
  nativeBuildInputs = [
    setuptools-scm
  ];

  buildInputs = [
    pytes
  ];
```

SuperSandro2000 commented:
@ofborg build python39Packages.allure-python-commons-test python39Packages.allure-python-commons python39Packages
